test/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2UnitTest.java

```diff
@@ -709,4 +710,39 @@ Response serve(IHTTPSession session) {
         //Then
         assertThat(httpMessagesSent, hasSize(greaterThan(0)));
     }
-}
+
+@Test
+public void shouldReportXssInJsonReponse() throws NullPointerException, IOException {
+    String test = "/shouldReportXssInJsonReponse/";
+    
+    this.nano.addHandler(new NanoServerHandler(test) {
+        @Override
+        Response serve(IHTTPSession session) {
+            String name = session.getParms().get("name");
+            String response;
+            if (name != null) {
+                response = getHtml("example.json",
+                        new String[][] {{"name", name}});
+            } else {
+                response = getHtml("NoInput.html");
```

Comment:
This could use the JSON file as well (with hardcoded name, or with the token as is).